tests/System.Buffers.Experimental.Tests/RangeTests.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Buffers.Tests
+{
+    public class RangeTests
+    {
+        [Fact]
+        public void Basics()
+        {
+
+            for (int first = -10; first < 10; first++)
+            {
+                for (uint length = 0; length < 20; length++)
+                {
+                    var range = new Range(first, length);
+
+                    Assert.Equal(first, range.First);
+                    Assert.Equal(length, range.Length);
+                    Assert.Equal(length + first, range.Last);
+
+                    long sum = 0;
+                    foreach (int value in range)
```

Comment:
Would you mind doing a micro-benchmark just to make sure that `foreach (var i in Range.Construct(i, j)) { ... }` is fast as we expect?

Reply:
will do